pkg/proxy/nftables/proxier.go

```diff
@@ -429,6 +450,18 @@ func (proxier *Proxier) setupNFTables(tx *knftables.Transaction) {
 		Rule:  "masquerade fully-random",
 	})
 
+	// Drop traffic to unused ClusterIPs.
+	if len(proxier.serviceCIDRs) > 0 {
+		tx.Add(&knftables.Rule{
+			Chain: KubeFilterPostRouting,
+			Rule: knftables.Concat(
+				ipX, "daddr", "{", proxier.serviceCIDRs, "}",
+				"drop",
+			),
+			Comment: ptr.To("Drop traffic to unused ClusterIPs"),
```

Comment:
1. Other kinds of drop/reject rules are installed in "regular chains", as opposed to "base chains", I wonder if this should follow the convention.
2. Given that we have to install reject rules in forward and output chains, is it better to install rules of "reject invalid ports of ClusterIPs" and "drop unused ClusterIPs" in the regular chain, to which "filter-forward" and "filter-output-post-dnat" will jump? Something like the following:
```
add rule ip kube-proxy filter-forward ct state new jump service-endpoints-check
add rule ip kube-proxy filter-forward ct state new jump clusterip-check

add rule ip kube-proxy filter-output-post-dnat ct state new jump clusterip-check

add rule ip kube-proxy clusterip-check ip daddr @cluster-ips reject
add rule ip kube-proxy clusterip-check ip daddr { 10.96.0.0/16 } drop
```
Also, it may be a bit more efficient to drop traffic to unused ClusterIPs earlier in forward or output hook.

cc @danwinship for his opinion.

Reply:
> Other kinds of drop/reject rules are installed in "regular chains", as opposed to "base chains", I wonder if this should follow the convention.

https://github.com/kubernetes/kubernetes/pull/122692#discussion_r1451580776

Reply:
I had suggested this approach above: https://github.com/kubernetes/kubernetes/pull/122692#discussion_r1451580776

Reply:
I like this idea of keeping cluster-ip checks together in one chain